LabSources/SampleScripts/Scenarios/CM-2002.ps1

```diff
@@ -0,0 +1,525 @@
+    [Switch]$AutoLogon
+)
+
+#region New-LabDefinition
+$NewLabDefinitionSplat = @{
+    Name                        = $LabName
+    DefaultVirtualizationEngine = "HyperV"
+    ReferenceDiskSizeInGB       = 100
+    ErrorAction                 = "Stop"
+}
+if ($PSBoundParameters.ContainsKey("VMPath")) { 
+    $Path = "{0}\{1}" -f $VMPath, $LabName
+    $NewLabDefinitionSplat["VMPath"] = $Path
+}
+New-LabDefinition @NewLabDefinitionSplat
+#endregion
+
+#region Initialise
+$PSDefaultParameterValues = @{
+    'Add-LabMachineDefinition:OperatingSystem' = $OSVersion
+    'Add-LabMachineDefinition:DomainName'      = $Domain
+    'Add-LabMachineDefinition:Network'         = $LabName
+    'Add-LabMachineDefinition:ToolsPath'       = "$labSources\Tools"
+    'Add-LabMachineDefinition:MinMemory'       = 1GB
+    'Add-LabMachineDefinition:Memory'          = 1GB
+}
+
+if ($AutoLogon.IsPresent) {
+    $PSDefaultParameterValues['Add-LabMachineDefinition:AutoLogonDomainName'] = $Domain
+    $PSDefaultParameterValues['Add-LabMachineDefinition:AutoLogonUserName']   = $AdminUser
+    $PSDefaultParameterValues['Add-LabMachineDefinition:AutoLogonPassword']   = $AdminPass
+}
+
+$DataDisk = "{0}-DATA-01" -f $CMHostname
+$SQLDisk = "{0}-SQL-01" -f $CMHostname
+
+$SQLConfigurationFile = "{0}\CustomRoles\CM-2002\ConfigurationFile-SQL.ini" -f $labSources
+#endregion
+
+#region Preflight checks
+switch ($true) {
+    (-not $SkipLabNameCheck.IsPresent) {
+        if ((Get-Lab -List -ErrorAction SilentlyContinue) -contains $_) { 
+            throw ("Lab already exists with the name '{0}'" -f $LabName)
+        }
+    }
+    (-not $SkipDomainCheck.IsPresent) {
+        try {
+            [System.Net.Dns]::Resolve($Domain) | Out-Null
+            throw ("Domain '{0}' resolves, choose a different domain" -f $Domain)
+        }
+        catch {
+            # resume
+        }
+    }
+    (-not $SkipHostnameCheck.IsPresent) {
+        foreach ($Hostname in @($DCHostname,$CMHostname)) {
+            try {
+                [System.Net.Dns]::Resolve($Hostname) | Out-Null
+                throw ("Host '{0}' resolves, choose a different hostname" -f $Hostname)
+            }
+            catch {
+                continue
+            }
+        }
+    }
+    # I know I can use ParameterSets, but I want to be able to execute this script without any parameters too, so this is cleaner.
+    ($PostInstallations.IsPresent -And $ExcludePostInstallations.IsPresent) {
+        throw "Can not use -PostInstallations and -ExcludePostInstallations together"
+    }
+    ($NoInternetAccess.IsPresent -And $PSBoundParameters.ContainsKey("ExternalVMSwitchName")) {
+        throw "Can not use -NoInternetAccess and -ExternalVMSwitchName together"
+    }
+    ($NoInternetAccess.IsPresent -And $Branch -eq "TP") {
+        throw "Can not install Technical Preview with -NoInternetAccess"
+    }
+    ((-not $NoInternetAccess.IsPresent) -And (Get-VMSwitch).Name -notcontains $ExternalVMSwitchName) { 
```

Comment:
```suggestion
    ((-not $NoInternetAccess.IsPresent) -And $ExternalVMSwitchName -eq 'Default Switch') { 
        break
    }
    ((-not $NoInternetAccess.IsPresent) -And (Get-VMSwitch).Name -notcontains $ExternalVMSwitchName) { 
```

This goes for the previous script as well - I would like to use my default switch and not create another adapter. This would enable such a scenario.